src/ucp/core/ucp_ep.c

```diff
@@ -157,6 +158,14 @@ ucs_status_t ucp_ep_create_base(ucp_worker_h worker, const char *peer_name,
 
 void ucp_ep_destroy_base(ucp_ep_h ep)
 {
+    ucs_assert(!(ep->flags & UCP_EP_FLAG_DESTROYED) && (ep->ref_cnt > 0));
+
+    if (--ep->ref_cnt != 0) {
+        return;
+    }
+
+    ep->flags |= UCP_EP_FLAG_DESTROYED;
```

Comment:
how does it work if ep is released right after?

Reply:
removed